bottlecap/src/telemetry/events.rs

```diff
@@ -0,0 +1,349 @@
+        #[serde(skip_serializing_if = "Option::is_none")]
+        version: Option<String>,
+    },
+
+    /// Record marking the completion of an invocation
+    #[serde(rename = "platform.runtimeDone", rename_all = "camelCase")]
+    PlatformRuntimeDone {
+        /// Request identifier
+        request_id: String,
+        /// Status of the invocation
+        status: Status,
+        /// When unsuccessful, the error_type describes what kind of error occurred
+        #[serde(skip_serializing_if = "Option::is_none")]
+        error_type: Option<String>,
+        /// Metrics corresponding to the runtime
+        #[serde(skip_serializing_if = "Option::is_none")]
+        metrics: Option<RuntimeDoneMetrics>,
+    },
+
+    /// Platfor report record
+    #[serde(rename = "platform.report", rename_all = "camelCase")]
+    PlatformReport {
+        /// Request identifier
+        request_id: String,
+        /// Status of the invocation
+        status: Status,
+        /// When unsuccessful, the error_type describes what kind of error occurred
+        #[serde(skip_serializing_if = "Option::is_none")]
+        error_type: Option<String>,
+        /// Metrics
+        metrics: ReportMetrics,
+    },
+
+    /// Extension-specific record
+    #[serde(rename = "platform.extension")]
+    PlatformExtension {
+        /// Name of the extension
+        name: String,
+        /// State of the extension
+        state: String,
+        /// Events sent to the extension
+        events: Vec<String>,
+    },
+
+    /// Telemetry processor-specific record
+    #[serde(rename = "platform.telemetrySubscription")]
+    PlatformTelemetrySubscription {
+        /// Name of the extension
+        name: String,
+        /// State of the extensions
+        state: String,
+        /// Types of records sent to the extension
+        types: Vec<String>,
+    },
+
+    /// Record generated when the telemetry processor is falling behind
+    #[serde(rename = "platform.logsDropped", rename_all = "camelCase")]
+    PlatformLogsDropped {
+        /// Reason for dropping the logs
+        reason: String,
+        /// Number of records dropped
+        dropped_records: u64,
+        /// Total size of the dropped records
+        dropped_bytes: u64,
+    },
+}
+
+/// Type of Initialization
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub enum InitType {
+    /// Initialised on demand
+    OnDemand,
+    /// Initialized to meet the provisioned concurrency
+    ProvisionedConcurrency,
+    /// SnapStart
+    SnapStart,
+}
+
+/// Phase in which initialization occurs
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum InitPhase {
+    /// Initialization phase
+    Init,
+    /// Invocation phase
+    Invoke,
+}
+
+/// Status of invocation/initialization
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum Status {
+    /// Success
```

Comment:
nit: not sure those comments are helpful?

Reply:
Agree, removing them